command-not-found.sh

```diff
@@ -0,0 +1,38 @@
+#!/bin/sh
+
+# for bash 4
+# this will be called when a command is entered
+# but not found in the user’s path + environment
+command_not_found_handle () {
+    toplevel=nixos # TODO: detect this somehow
+    cmd=$1 # TODO: differentiate between paths and commands
+    attrs=$(@out@/bin/nix-locate -1 --no-group --type x --top-level --regex "/bin/$cmd$")
+    len=$(echo -n "$attrs" | grep -c "^")
+
+    case $len in
+        0)
+            >&2 echo "$cmd: command not found"
+            ;;
+        1) # TODO: add option to autorun with 1 match
+            >&2 echo "The program '$cmd' is currently not installed. You can install it"
+            >&2 echo "by typing:"
+            >&2 echo "  nix-env -iA $toplevel.$attrs"
+            ;;
+        *)
+            >&2 echo "The program '$cmd' is currently not installed. It is provided by"
+            >&2 echo "several packages. You can install it by typing one of the following:"
+            for attr in $(echo $attrs); do # HACK: fix me
```

Comment:
I think "echo" is probably be the simplest solution here, but not knowing bash very well I have to say it looks weird.